### DISCUSSION TITLE
While finetune tts_models--multilingual--multi-dataset--xtts_v2, but error happen,how to fix it? Thanks in advance!

### DISCUSSION THREAD

**Discussion by jackyin68** [coqui-ai/TTS#3221] [General]
Whether this problem have been fixed which has happened but fixed because of not supported xtts finetuning.
----------------------------------------------------------------------------------------------------------------------------

Traceback (most recent call last):
  File "D:\Projects\TTS\TTS\bin\train_tts.py", line 71, in <module>
    main()
  File "D:\Projects\TTS\TTS\bin\train_tts.py", line 58, in main
    trainer = Trainer(
  File "D:\Projects\TTS\venv\lib\site-packages\trainer\trainer.py", line 502, in __init__
    self.criterion = self.get_criterion(self.model)
  File "D:\Projects\TTS\venv\lib\site-packages\trainer\trainer.py", line 2077, in get_criterion
    criterion = model.get_criterion()
  File "D:\Projects\TTS\venv\lib\site-packages\torch\nn\modules\module.py", line 1695, in __getattr__
    raise AttributeError(f"'{type(self).__name__}' object has no attribute '{name}'")
AttributeError: 'Xtts' object has no attribute 'get_criterion'

**Comment by todochenxi** [comment]
I'm also experiencing this problem, if there is a solution, please @ me, thank you very much!

**Comment by jackyin68** [comment]
Project member told me to Use ..\TTS\recipes\ljspeech\xtts_v2\train_gpt_xtts.py.  Maybe it can work.

You can finetune firstly. About infer, you should make some modification.

**Reply by mshbaita-jo** [reply]
in which model and where can I find this path?

**Comment by todochenxi** [comment]
不幸的是，我用的就是这个脚本，hhhh，
Unfortunately, that's the script I'm using. hhhh

**Comment by jackyin68** [comment]
你的报错是什么
Give the error in more detail.

**Reply by todochenxi** [reply]
看错了，两个脚本我都试了，bin 下得脚本报得错和你一样，然后 /recipes/ljspeech/xtts_v2/train_gpt_xtts.py 这个报得错是`AssertionError:  ❗ len(DataLoader) returns 0. Make sure your dataset is not empty or len(dataset) > 0.`

**Reply by todochenxi** [reply]
https://github.com/coqui-ai/TTS/issues/3229 和这个差不多

**Comment by jackyin68** [comment]
检查一下配置信息，以及dataset的格式。 check config and data format.
==============================================
config_dataset = BaseDatasetConfig(
    formatter="ljspeech",
    dataset_name="dsz_dataset",
    path="D:/Projects/TTS/dataset/dsz_dataset/",
    meta_file_train="D:/Projects/TTS/dataset/dsz_dataset/metadata.csv",
    language="zh-cn",
)

**Reply by todochenxi** [reply]
![图片](https://github.com/coqui-ai/TTS/assets/108149467/b4f71961-0800-42eb-b330-3171665039ee)Yes, I checked and there is no problem, but I am not sure about the settings here:
![Uploading 图片.png…]()

**Comment by jackyin68** [comment]
the dir wavs remove and try again. only ../noah
